src/control/server/mgmt_system.go

```diff
@@ -712,3 +713,45 @@ func (svc *mgmtSvc) LeaderQuery(ctx context.Context, req *mgmtpb.LeaderQueryReq)
 	svc.log.Debugf("MgmtSvc.LeaderQuery dispatch, resp:%+v\n", resp)
 	return resp, nil
 }
+
+// ClusterEvent management service gRPC handler receives ClusterEvent requests
+// from control-plane instances attempting to notify the MS of a cluster event
+// in the DAOS system.
+//
+// On receipt of the cluster event request, process any encapsulated RAS event
+// messages (handling of other types to be supported in the future).
+//
+// Necessary state updates for the rank involved in the event will be
+// registered in the system membership.
+func (svc *mgmtSvc) ClusterEvent(ctx context.Context, req *mgmtpb.ClusterEventReq) (*mgmtpb.ClusterEventResp, error) {
+	if req == nil {
+		return nil, errors.New("nil request")
+	}
+	if req.Sequence < 1 {
+		return nil, errors.New("invalid sequence number in request")
+	}
+	svc.log.Debugf("MgmtSvc.ClusterEvent dispatch, req:%#v\n", req)
+
+	if err := svc.sysdb.CheckLeader(); err != nil {
+		return nil, err
+	}
+
+	rasEventPB := req.GetRas()
+	if rasEventPB == nil {
+		return nil, errors.Errorf("unexpected event type received, want RAS got %T",
+			req.GetEvent())
+	}
+
+	rasEvent := new(ras.Event)
+	if err := convert.Types(rasEventPB, rasEvent); err != nil {
+		return nil, errors.Wrap(err, "convert proto to ras event")
+	}
+	if err := ras.ProcessEvent(svc.log, rasEvent); err != nil {
+		return nil, err
+	}
```

Comment:
Yeah, so here I think you would want to feed the event into your event processor entity that was configured as part of setting up the server. This would make testing easy as well, as a processor with no configured handlers just ignores events.

Reply:
will look into it